test/assertEx.ts

```diff
@@ -0,0 +1,117 @@
+import * as assert from 'assert';
+import { parseError } from 'vscode-azureextensionui';
+
+function areUnorderedArraysEqual<T>(actual: T[], expected: T[]): { areEqual: boolean, message?: string } {
+    actual = actual.slice();
+    expected = expected.slice();
+    actual.sort();
+    expected.sort();
+
+    let message = `Actual:   ${JSON.stringify(actual)}\nExpected: ${JSON.stringify(expected)}`
+
+    if (!(actual.length === expected.length)) {
+        return { areEqual: false, message };
+    }
+
+    for (let i = 0; i < actual.length; ++i) {
+        if (actual[i] !== expected[i]) {
+            return { areEqual: false, message };
+        }
+    }
+
+    return { areEqual: true };
+}
+
+export function unorderedArraysEqual<T>(actual: T[], expected: T[], message?: string): void {
+    let result = areUnorderedArraysEqual(actual, expected);
+    assert(result.areEqual, `${message || "Unordered arrays are not equal"}\n${result.message}`);
+}
+
+export function notUnorderedArraysEqual<T>(actual: T[], expected: T[], message?: string): void {
+    let result = areUnorderedArraysEqual(actual, expected);
+    assert(!result.areEqual, `${message}\n${result.message}`);
+}
```

Comment:
Question: Would it make sense to use have unorderedArraysEqual return a boolean, then use `console.assert(!unorderedArraysEqual)`. It looks like you're defining a second test B to solely test the negation of test A.

Reply:
> It looks like you're defining a second test B to solely test the negation of test A.

It's not a test, it's an assertion to be used in tests. Having separate functions like this allows both to provide helpful messages for analyzing tests.  That didn't happen correctly here, though, fixed the message:

    assert(!result.areEqual, `${message || "Unordered arrays are equal but were expected not to be"}\n${result.message}`);